language/move-prover/src/prelude.bpl

```diff
@@ -367,35 +367,12 @@ function {:inline} $InRange(r: Value, i: int): bool {
 type {:datatype} Location;
 function {:constructor} Global(t: TypeValue, a: int): Location;
 function {:constructor} Local(i: int): Location;
+function {:constructor} Param(i: int): Location;
```

Comment:
I just figured out what was going on with Param here.  This is a trick for saying that N things are disjoint without N choose 2 assertions xi != xj.  The Param is a constructor, so Param(i) != Param(j) if i != j. Is there a concern about consistency, since the location really is a Local or Global and those are disjoint from Param's?

Reply:
I don't think there is any concern about consistency, because the actual shape of a reference parameter is opaque to a function. The code cannot reflect over whether it is a local, a global, or whatsoever reference as a parameter. This is actually the idea also underlying the new memory model where (a) we can replace immutable refs by values (b) can transform mutable refs to in/out parameters. In any case, appreciate if Shaz can confirm that I did not make a mistake here.